hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -2085,6 +2085,10 @@ public void move(final byte[] encodedRegionName, byte[] destServerName) throws I
         return;
       }
     }
+    if (dest.equals(LoadBalancer.BOGUS_SERVER_NAME)) {
```

Comment:
Since this is not merged, could you add more explanation about it?

Reply:
When rsgroup is enabled a region can be moved to same rsgroup servers only. It means that region's current server and target server must belong to the same rsgroup.

Currently while moving a region to different rsgroup server, balancer assigns invalid target server named as BOGUS_SERVER_NAME (localhost,1,1.). Even though it is invalid server region move procedure is submitted and finally region move fails.

This PR proposes to fail fast, in case of target server is BOGUS_SERVER_NAME log info and return. No need to submit region move procedure

Reply:
I mean add more annotate in the code, not to me only...

Reply:
@arshadmohammad  Can you please add code comment to understand the change better.

Reply:
Added comment in code.
Force pushed to correct author name in previous commit